docs/docs/migration-guide.mdx

```diff
@@ -24,6 +26,916 @@ If you still have training data in Markdown format then the recommended approach
+          self._rule_only_data = kwargs.get("rule_only_data", {})
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {6}
+  class MyPolicy(Policy):
+      def predict_action_probabilities(
+          self,
+          tracker: DialogueStateTracker,
+          domain: Domain,
+          rule_only_data: Optional[Dict[Text, Any]] = None,
+    ) -> PolicyPrediction:
+  ```
+
+  </TabItem>
+</Tabs>
+
+#### Changes to Custom NLU Components
+
+**Inheriting from `GraphComponent`**
+
+NLU components which previously inherited from one of the following classes additionally
+need to inherit from the
+[`GraphComponent` interface](custom-graph-components.mdx#the-graphcomponent-interface):
+- `SparseFeaturizer`
+- `DenseFeaturizer`
+- `IntentClassifier`
+- `EntityExtractor`
+- `Component`
+
+This snippet shows the required changes:
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-nlu-parent-class" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyNLUComponent(SparseFeaturizer):
+      ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {3}
+  from rasa.engine.graph import GraphComponent
+
+  class MyNLUComponent(GraphComponent, SparseFeaturizer):
+      ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+** Inheriting from `EntityExtractorMixin` instead of `EntityExtractor`**
+
+The `EntityExtractor` class was renamed to `EntityExtractorMixin`:
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-nlu-entity-extractor" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  from rasa.nlu.extractors.extractor import EntityExtractor
+
+  class MyNLUComponent(EntityExtractor):
+      ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {2,4}
+  from rasa.engine.graph import GraphComponent
+  from rasa.nlu.extractors.extractor import EntityExtractorMixin
+
+  class MyNLUComponent(GraphComponent, EntityExtractorMixin):
+      ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+
+**Instantiating a NLU Component for Training**
+
+NLU components are no longer instantiated via their constructor. Instead, all NLU
+components have to override the `create` method of the
+[`GraphComponent` interface](custom-graph-components.mdx#the-graphcomponent-interface). The
+passed in configuration is your NLU component's default configuration including any updates
+from your model configuration file.
+
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-nlu-init" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyNLUComponent(IntentClassifier):
+
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
+        super().__init__(component_config)
+      ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python
+  class MyNLUComponent(GraphComponent, IntentClassifier):
```

Comment:
For completeness, does it make sense to also add the imports for the types? Maybe the registry decorator as well?

Reply:
added imports everywhere